### DISCUSSION TITLE
Slow query when manually connect the PSQL to a custom Golang API

### DISCUSSION THREAD

**Discussion by febriliankr** [supabase/supabase#1297] [Questions] [answered]
I tried to directly use the PSQL database provided by Supabase (through Connection String in Settings), made a Go app to query from one table. There are only 5 rows, and I queried only one column. I found out that the query is extremely slow, it's always above 1500 ms. 

<img width="1013" alt="Screen Shot 2021-04-28 at 17 37 43" src="https://user-images.githubusercontent.com/65109548/116391000-d5063e80-a848-11eb-8030-01bd8aff7a06.png">

The app is in localhost so it's definitely caused by the PSQL database hosted by Supabase. Is this normal to get above 1500 ms query response every time? 

Thank you very much

Note: If anyone wants to see the code it's here: https://gist.github.com/febriliankr/0a89f6ed83037d3da4631c458b0e51dd
Just a standard go code

**Comment by steve-chavez** [comment, marked as answer]
Hm. Maybe it's because you're opening a new PostgreSQL connection every time you do the request?

Try with plain `psql`(the command) and enable timing to see the actual response time. Check this [previous discussion](https://github.com/supabase/supabase/discussions/577#discussioncomment-302817) for how to do that.

**Edit**: `psql` responds with adequate time. The issue is in the Golang server.

**Reply by febriliankr** [reply]
Yup, I got around 263 ms when querying directly from PSQL. But the Golang code has never been a problem with another remote database, that Go API is usually responding at around 300 ms to a remote DB. Plus, I tried reusing open DB connection and still got around 1200-1300 ms.

![image](https://user-images.githubusercontent.com/65109548/116437406-a05dab80-a877-11eb-9440-da5f4836cf4d.png)

What am I still missing here? Thanks

**Reply by steve-chavez** [reply]
> Yup, I got around 263 ms when querying directly from PSQL.

Hm. I don't know Golang so I cannot be of help there, sorry :disappointed:. But from your statement, it's clear that the issue is not on Supabase's end - `psql` query time is good.
